features/old/applications/providers/create_application.feature

```diff
@@ -134,6 +133,58 @@
       And should see "Application was successfully created"
       And  buyer "bob" should have 1 cinstance
 
+  Scenario: Create an application with Service Plans disabled from Account context
+    Given plan "Basic" is customized
+    And buyer "bob" has no applications
+    And provider "foo.3scale.localhost" has "service_plans" switch denied
+    And I go to the account context create application page for "bob"
+    Then I should see "New Application"
+    And I should not see "Service Plan"
+    And I select "API" from "Product"
+    But the application plans select should not contain custom plans of provider "foo.3scale.localhost"
+    And I select "Basic" from "Application plan"
+    And I fill in "Name" with "UltimateWidget"
+    And I fill in "Description" with "Awesome ultimate super widget"
+    And I press "Create Application"
+    Then I should be on the provider side "UltimateWidget" application page
+    And should see "Application was successfully created"
+    And buyer "bob" should have 1 cinstance
+
+  Scenario: Create an application with Service Plans disabled from Product context
+    Given plan "Basic" is customized
+    And provider "foo.3scale.localhost" has "service_plans" switch denied
+    And buyer "bob" has no applications
+    And I go to the product context create application page for "API"
+    Then I should see "New Application"
+    But I should not see "Service plan"
+    And I select "bob" from "Account"
+    But the application plans select should not contain custom plans of provider "foo.3scale.localhost"
+    And I select "Basic" from "Application plan"
+    And I fill in "Name" with "UltimateWidget"
+    And I fill in "Description" with "Awesome ultimate super widget"
+    And I press "Create Application"
+    Then I should be on the provider side "UltimateWidget" application page
+    And should see "Application was successfully created"
+    And buyer "bob" should have 1 cinstance
+
+  Scenario: Create an application with Service Plans disabled from Audience context
+    Given plan "Basic" is customized
+    And buyer "bob" has no applications
+    And provider "foo.3scale.localhost" has "service_plans" switch denied
+    And I go to the audience context create application page
+    Then I should see "New Application"
+    But I should not see "Service plan"
+    And I select "bob" from "Account"
+    And I select "API" from "Product"
+    But the application plans select should not contain custom plans of provider "foo.3scale.localhost"
+    And I select "Basic" from "Application plan"
+    And I fill in "Name" with "UltimateWidget"
+    And I fill in "Description" with "Awesome ultimate super widget"
+    And I press "Create Application"
+    Then I should be on the provider side "UltimateWidget" application page
+    And should see "Application was successfully created"
+    And buyer "bob" should have 1 cinstance
+
```

Comment:
I'd say just do not write any more cucumber like this. Remove that and redo it in another PR

Reply:
It's a terrible test but it's still a test and the bug must have one. I'd rather leave it here and refactor the whole file later.

I can offer adding an integration test instead of this cucumber, but that would take more time. I'd really keep these scenarios **TEMPORARILY** and redo the whole file in another PR.

Reply:
The issue is that we never redo them Remember your last PRs ... :)